docs/assignment3.md

```diff
@@ -1,110 +1,776 @@
+
+***GUI Implementation***\
+For the GUI implementation, we decided to go with `SWING`. \
+Since we will have three main views we split the GUI into three forms. Every form contains the needed components mentioned in the class-diagram. All of the GUI classes have an instance to the `GlobalsView`. This "connection" will provide the Views to create, style, etc. and also use its functions. Every View class has a `JPanel` called `mainPanel`. This panel is used every time the user is changing the screen. The `State` is calling this `mainPanel` and replacing the current `mainPanel` with the requested one. Moreover, the `currentState` in the `Globals` class will be the variable telling us where the application is at the moment.
+
+***State, Globals, GlobalsViews***\
+After some discussion, we decided to create these classes as our "helpers". The classes are crucial for our application. \
+`State` class is changing the state and screen of the application. (The class is explained in GUI Implementation part)
+
+The `Globals` and `GlobalsView` are the classed that provide and control the data for the whole system. In `Globals` we can get and use the data from the `settings.json` and `snippets.json` files. Using these classes we will gain access to a single instance if the `settingsParser` and `snippetParser`. Furthermore, the `Globals` also contains a `SnippetHelper` that will complete the snippet manipulation requests. 
+
+***Using Design Patterns***\
+Using design patterns helped us improve our code and make it more robust. As explained in the "Design Pattern" we were worried that by accident we can have more than one object which was going to cause errors. Using the singleton was the best solution for our problem. More over, since we could not include the other two design patterns our code end up to not be so pretty but functional.
+
+***
+The main fail of the application is located in:\
+/src/main/java/Main.java
+
+
+The JAR file is located in:\
+out/artifacts/project_jar/project.jar
+
+**Here are the instructions on how to start our application:**\
+1. Create a folder in convenient place for you or use an existing one
+2. Copy the full path to the folder in step 1
+3. Open Terminal/PowerShell
+4. Navigate to the folder where you have download the `project.jar` file (can be the same folder as the one in step 1)
+5. Run the following command:\
+`java -cp project.jar Main --path "<FULL FOLDER PATH>"`\
+or\
+`java -cp project.jar Main -p "<FULL FOLDER PATH>"`<br><br>
+` <FULL FOLDER PATH>` is the place where you need to paste the full path you copied in step 2.
+
+***Video***\
+![Plase see our live demo (if video do not load it can be downloaded from the repository)](./video/demo_assignment3.mp4)
```

Comment:
Very good implementation, also well aligned with your UML models. Java classes follow the basic rules by Ousterhout and Robillard. Just some big classes are present. The code is very neat and well readable, although some parts are quite complex. You implemented most features described in assignment 1.